ipatests/test_integration/test_automember.py

```diff
@@ -0,0 +1,366 @@
+        addition of user entry
+        """
+        self.master.run_command(['ipa', 'user-add', '--first=%s' % first,
+                                 '--last=%s' % last, user,
+                                 '--setattr=destinationindicator=%s' % group])
+
+    def user_automember_mod(self, group, user):
+        """
+        modfication of user entry
+        """
+        self.master.run_command(['ipa', 'user-mod', user,
+                                 '--setattr=destinationindicator=%s' % group])
+
+    def user_automember_cleanup(self, user):
+        """
+        deletion of user entry
+        """
+        self.master.run_command(['ipa', 'user-del', user])
+
+    def automember_group_cleanup(self, group):
+        """
+        cleanup of automember/group entry
+        """
+        cmdlist = [['ipa', 'automember-del', '--type=group', group],
+                   ['ipa', 'group-del', group]]
+
+        for cmd in cmdlist:
+            self.master.run_command(cmd, raiseonerr=False)
+
+    def automember_hostgroup_add(self, hostgroup):
+        """
+        addition of hostgroup/automember entry
+        """
+        cmdlist = [['ipa', 'hostgroup-add', '--desc=%s' % hostgroup,
+                   hostgroup],
+                   ['ipa', 'automember-add', '--type=hostgroup', hostgroup],
+                   ['ipa', 'automember-add-condition',
+                    '--key=nshostlocation', '--type=hostgroup',
+                    '--inclusive-regex=%s' % hostgroup, hostgroup]]
+
+        for cmd in cmdlist:
+            self.master.run_command(cmd)
+
+    def host_automember_setup(self, hostgroup, host):
+        """
+        addition of host entry
+        """
+        self.master.run_command(['ipa', 'host-add', host, '--location=%s'
+                                 % hostgroup, '--force'])
+
+    def host_automember_mod(self, hostgroup, host):
+        """
+        modfication of host entry
+        """
+        self.master.run_command(['ipa', 'host-mod', host, '--location=%s'
+                                 % hostgroup])
+
+    def host_automember_cleanup(self, host):
+        """
+        modfication of host entry
+        """
+        self.master.run_command(['ipa', 'host-del', host])
+
+    def automember_hostgroup_cleanup(self, hostgroup):
+        """
+        cleanup of automember/hostgroup entry
+        """
+        cmdlist = [['ipa', 'automember-del', '--type=hostgroup', hostgroup],
+                   ['ipa', 'hostgroup-del', hostgroup]]
+
+        for cmd in cmdlist:
+            self.master.run_command(cmd, raiseonerr=False)
+
+    def test_modify_user_entry_unmembership(self):
+        # https://pagure.io/389-ds-base/issue/50077
+        """
+        Test that an update on a user entry triggers a re-evaluation of the
+        user's automember groups when auto-unmembership is active.
+        """
+        try:
+            tasks.kinit_admin(self.master)
+            master = self.master
+            search_cmd = [
```

Comment:
Instead of calling ldapsearch on master it is better to use ldap client provided by  master.ldap_connect.
You can see example of usage in tasks.wait_for_replication function.